vector/src/main/scala/geotrellis/vector/Polygon.scala

```diff
@@ -38,34 +37,44 @@ object Polygon {
   def apply(exterior: Line): Polygon =
     apply(exterior, Set())
 
+  def apply(exterior: CoordinateSequence): Polygon =
+    Polygon(factory.createPolygon(exterior))
+
   def apply(exterior: Line, holes:Line*): Polygon =
     apply(exterior, holes)
 
-  def apply(exterior: Line, holes:Traversable[Line]): Polygon = {
+  def apply(exterior: CoordinateSequence, holes: GenTraversable[CoordinateSequence]): Polygon =
+    apply(factory.createLinearRing(exterior), holes.map(factory.createLinearRing))
+
+  def apply(exterior: Line, holes: GenTraversable[Line]): Polygon = {
     if(!exterior.isClosed) {
       sys.error(s"Cannot create a polygon with unclosed exterior: $exterior")
     }
 
     if(exterior.vertices.length < 4) {
-      sys.error(s"Cannot create a polygon with exterior with less that 4 points: $exterior")
+      sys.error(s"Cannot create a polygon with exterior with fewer than 4 points: $exterior")
```

Comment:
Using this PR to think about immutability vs performance in `geotrellis.vector` and noticed that ^ in line above where `. vertices.length` is called an `Array[Point]` is created to perform the length call. `.vertexCount` should be safe substitute.